src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala

```diff
@@ -4,30 +4,42 @@ package net.cakesolutions
 
 import scala.util._
 
-import sbt._
-import sbt.IO._
 import sbt.Keys._
+import sbt._
+import sbtbuildinfo.BuildInfoPlugin
+import sbtbuildinfo.BuildInfoPlugin.autoImport._
 
-import sbtbuildinfo.BuildInfoPlugin, BuildInfoPlugin.autoImport._
-
-// kicks in if the Project uses the BuildInfoPlugin
+/**
+  * Assumes that `git` is installed. Used to configure and load the `BuildInfo` object at runtime. This
+  * object contains information regarding the build such as the git commit hash, etc.
+  *
+  * Configuration of this plugin should be avoided in local project SBT build files.
+  */
 object CakeBuildInfoPlugin extends AutoPlugin {
-  override def requires = BuildInfoPlugin
-  override def trigger = allRequirements
+  require(
+    Try("git --version".!!).isSuccess,
```

Comment:
True, but git not being present is also a bad idea - hence why we should handle this error scenario. Either way, SBT needs to stop here, so what is the best way of handing this type of error scenario?

We can sys.exit, throw an exception or appeal to some SBT error handling.

I'm not immediately aware how to handle SBT errors in this type of context, hence why I prefer an exception trace over a sys.exit.